independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Methods.java

```diff
@@ -197,17 +197,31 @@ static boolean isObjectToString(MethodInfo method) {
                 }
             }
             if (!merged.isEmpty()) {
-                boolean addToCandidates = true;
-                if (Modifier.isFinal(method.flags())) {
-                    if (transformUnproxyableClasses) {
-                        methodsFromWhichToRemoveFinal.add(NameAndDescriptor.fromMethodInfo(method));
+                if (Modifier.isPrivate(method.flags())) {
+                    if (merged.size() == 1) {
+                        LOGGER.warn(String.format(
```

Comment:
How about the `Logger.warnf()` method instead of `String.format()`?

Reply:
Good idea, done